geode-client-protocol/src/main/java/org/apache/geode/internal/protocol/operations/OperationHandler.java

```diff
@@ -30,7 +31,11 @@
   /**
    * Decode the message, deserialize contained values using the serialization service, do the work
    * indicated on the provided cache, and return a response.
+   *
+   * @throws ConnectionStateException if the connection is in an invalid state for the operation in
+   *         question.
    */
   Result<Resp, ErrorResp> process(SerializationService serializationService, Req request,
-      MessageExecutionContext messageExecutionContext) throws InvalidExecutionContextException;
+      MessageExecutionContext messageExecutionContext)
+      throws InvalidExecutionContextException, ConnectionStateException;
```

Comment:
Is `ConnectionStateException` still thrown? Or is the `throws` just inherited from `ConnectionStateProcessor`?

Reply:
ConnectionStateException is newly thrown in this change, Galen made the observation that the code we're using to handle this inside the OpHandler is identical to the catch code we're using in the function calling this.